### ISSUE TITLE
Clr Date Picker transforms and deletes date on mobile devices

### ISSUE THREAD

**Issue by martinbrom** [vmware/clarity#3961]
## Describe the bug

When using a mobile device to input date using NATIVE mobile datepicker, the date gets converted by clarity into angular date format, which is incorrect according to https://tools.ietf.org/html/rfc3339, displays a warning in the console and deletes the input value.

## How to reproduce

![datepickerissue](https://user-images.githubusercontent.com/16322387/67952173-2baba180-fbed-11e9-98a3-4889dda38f30.gif)

**THIS ISSUE IS ONLY REPRODUCABLE ON ACTUAL MOBILE DEVICES (FOR ME)**

1.  Go to clarity date picker documentation
2.  Click on Code & Examples
3.  Click on Date Picker input field under Basic API
4.  Input any date using native mobile Date Picker
5.  Unexpected behavior happens:
5a.  Date in the input gets deleted and warning is printed into the console (The specified value "11/01/2019" does not conform to the required format, "yyyy-MM-dd")
5b.  I think this could be solved by swapping the order of first two condition branches in the updateInput() function in the forms/datepicker/date-input.ts file.

## Expected behavior

When using a mobile device to input date using NATIVE mobile datepicker, the date should be visible in the input field.

## Versions

**App**

* Angular: 8.0.2
* Node: 10.15.3
* Clarity: 2.2.0

**Device:**

* Type: Honor 8
* OS: Android 7.0
* Browser: Chrome (Mobile)
* Version: 74.0.3729.157

## Additional notes

https://github.com/vmware/clarity/issues/3648
This issue was probably mentioning the same thing but got closed without consideration.

**Comment by ElesGergo** [comment]
I ran into the same issue.
Tested on ios.
Any updates regarding the issue?

**Comment by gnomeontherun** [comment]
This seems like a disconnect between the format of the native controls and the website locale. I suspect this works fine if you use a US-EN locale on your phone while using our documentation.

You should set the locale for the language that you are targetting to get this to work. I will leave this open as we need to identify a better solution here that isn't so coupled to Angular locales.

**Comment by martinbrom** [comment]
I tried changing my locale to en_US and I still get the same issue.
![datepickerissue2](https://user-images.githubusercontent.com/16322387/68123722-029d5080-ff0e-11e9-8b7c-adafd18e2fe5.gif)

When trying a different date-picker on the page with the changed locale, this one seems to work almost fine. The only difference is the -1 day difference between the date I input and the date that is shown. This is more than likely a timezone issue, since I am currently in GMT+1 and the result value of the input is -1 day 23:00. I suppose this means the timezone gets subtracted from the date I select, taking 00:00 as input time.
![datepickerissue3](https://user-images.githubusercontent.com/16322387/68123725-07fa9b00-ff0e-11e9-9dd7-463d1693e6ca.gif)

**Comment by martinbrom** [comment]
I also have some further findings regarding the issue:

The basic API datepicker demo uses this code and the issue is reproducible there:
```html
<form class="datepicker-demo" clrForm clrLayout="vertical">
    <clr-date-container>
        <input type="date" clrDate name="demo" [(ngModel)]="demo">
    </clr-date-container>
</form>
```

The en_US datepicker demo uses this code and the code works fine there:
```html
<form clrForm clrLayout="vertical">
    <input type="date" clrDate>
</form>
```

Is there a reason why these two demos use a different code?

It would seem that either the clr-date-container or the input binding is the culprit behind the issue. I will fork the repository and try testing locally and will get back in the upcoming days with further findings.

**Comment by martinbrom** [comment]
I would like to fix this bug. Proposal template follows:

## Summary

* What is the change?

Allow usage of datepicker component on mobile devices.
When using a native datepicker, date will now not be formatted into application locale, since input element of type="date" needs a format of "yyyy-mm-dd" on mobile devices. Value of the input will get set directly (planned) instead of being formatted first and then set (current).

* Does this change impact existing behaviors? If so how?

Datepicker on mobile devices will now correctly set selected date. As described above, currently the input value gets deleted when selecting a date under specific conditions.

## Implementation Plan

Swap if and elseif blocks order, the check if you are using a mobile device should have precedence over input having a control.
* What parts of the code are affected?

clr-angular/forms/datepicker/date-input.ts : updateInput()

* Will you need to refactor existing Clarity code?

Yes

## Conclusion

Implementation consists only of swapping two blocks of code -> almost no time needed at all.

**Comment by gnomeontherun** [comment]
@martinbrom Please go for it, thanks for looking into this and helping out!

**Comment by martinbrom** [comment]
@gnomeontherun Could you please create a topic branch so I can submit a PR with the fix?

**Comment by gnomeontherun** [comment]
@martinbrom you don't need a topic branch for a bugfix, that is usually for something larger we work on for a longer timeline. Thanks!

**Comment by hippee-lee** [comment]
This is closed by #3992
